File: customtablewidget.cpp
#include "customTableWidget.h"

#include <QWidgetItem>

customTableWidget::customTableWidget( QWidget *parent )
: QTableWidget( parent )
{
    setSelectionMode(QAbstractItemView::NoSelection);

    connect( this, SIGNAL( clicked( const QModelIndex& ) ),
          this, SLOT( selectRow( const QModelIndex& ) ));
}

//--------------------------------------------------------------------

customTableWidget::~customTableWidget()
{
     this->QTableWidget::~QTableWidget();
}

//--------------------------------------------------------------------

void customTableWidget::editItem( QTableWidgetItem * item)
{
    //do nothing
}

//--------------------------------------------------------------------

void customTableWidget::selectRow( const QModelIndex& index )
{
     //do nothing
}

File: custom_tableWidget.py
from PyQt4 import QtCore, QtGui

class CustomTableWidget(QtGui.QTableWidget):
    def __init__(self, parent = None):
        super(CustomTableWidget, self).__init__(parent)
        self.setSelectionMode(QtGui.QAbstractItemView.NoSelection)
        self.setFocusPolicy(QtCore.Qt.NoFocus)

    def selectRow(self, p_int):
       #do nothing
       return

    def editItem(self, QTableWidgetItem):
        # do nothing
        return
